UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -44,15 +44,15 @@ public class UnitsNetJsonConverter : JsonConverter
             object obj = TryDeserializeIComparable(reader, serializer);
             if (obj is Array values)
             {
-                // Create array with the requested type, such as `Length[]` or `Frequency[]`
-                var arrayOfQuantities = Array.CreateInstance(objectType.GetElementType(), values.Length);
+
+                // Create array with the requested type, such as `Length[]` or `Frequency[]` or multi-dimensional arrays like `Length[,]` or `Frequency[,,]` 
+                var arrayOfQuantities = Array.CreateInstance(objectType.GetElementType(), LastIndex(values));
 
                 // Fill array with parsed quantities
-                var i = 0;
-                foreach (ValueUnit valueUnit in values)
+                var ind = FirstIndex(values);
```

Comment:
> Nitpick, but I think a more common naming convention is `idx` or `index`.

Fixed.

> Also use explicity type instead of `var`, it was not clear to me at all that this was not simply an integer and I didn't immediately see why the while-loop below didn't run forever.

Also changed, Visual Studio is yelling a me to use 'var' haha, but I agree it is confusing at first glance because of the previously discussed concept of 'index' when referring to multidimensional arrays.